src/java/ognl/security/MethodBodyExecutionSandbox.java

```diff
@@ -0,0 +1,87 @@
+package ognl.security;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+/**
+ * Guarantee a singleton shared thread-safe security manager and sandbox for user's methods body execution.
+ *
+ * Add the `-Dognl.security.manager` to JVM options to enable.
+ *
+ * <p> Note: Due to potential performance and concurrency issues, try this only if you afraid your app can have an
+ * unknown "expression injection" flaw or you afraid you cannot prevent those in your app's internal sandbox
+ * comprehensively e.g. you cannot discover and maintain all attack vectors over time because of many dependencies
+ * and also their change over time.</p>
+ *
+ * <p> This tries to provide an option to you to enable a security manager that disables any sensitive action e.g.
+ * exec and exit even if attacker had a successful "expression injection" in any unknown way into your app. However,
+ * also honors previous security manager and policies if any set, as parent, and rolls back to them after method
+ * execution finished.</p>
+ *
+ * @author Yasser Zamani
+ * @since 3.1.23
+ */
+public class MethodBodyExecutionSandbox {
+    private static int residentsCount = 0;
+
+    private static SecurityManager parentSecurityManager;
+
```

Comment:
To minimize overhead with `System.getProperty("ognl.security.manager")` calls and to avoid possible SecurityException at unexpected times, maybe consider setting a one-time-flag using a static initializer ?  Something like:
```java
    private static final boolean sandboxEnabled;
    static {
        boolean tempSandboxEnabled = false;
        try {
            tempSandboxEnabled = (System.getProperty("ognl.security.manager") != null);
        } catch (Exception ex) {
            // Ignore property read failure
        } finally {
            sandboxEnabled = tempSandboxEnabled;
        }
    }
``` 
and refer to that sandboxEnabled flag instead.  Unfortunately it makes unit testing more complicated.

Reply:
I would like to keep this check lazy to allow user enable/disable it programmatically but it's a good idea to surround `getProperty` with try/catch to honor upper level security manager, I'll 👍